test/jest/victory-core/victory-util/data.test.js

```diff
@@ -0,0 +1,382 @@
+/* eslint no-unused-expressions: 0, max-nested-callbacks: 0 */
+import React from "react";
+import { Data, VictoryPortal } from "victory-core";
+import { fromJS } from "immutable";
+
+const immutableDataTest = {
+  createData: (data) => fromJS(data),
+  testLabel: "data in immutable"
+};
+
+const dataTest = {
+  createData: (data) => data,
+  testLabel: "data in js"
+};
+
+describe("victory-util/data", () => {
+  describe("createStringMap", () => {
+    const tickValues = ["one", "two", "three"];
+    const categories = ["red", "green", "blue"];
+
+    it("returns a string map from strings in tickValues", () => {
+      const props = { tickValues };
+      const stringMap = Data.createStringMap(props, "x");
+      expect(stringMap).toEqual({ one: 1, two: 2, three: 3 });
+    });
+
+    it("returns a string map from strings in categories", () => {
+      const props = { categories };
+      const stringMap = Data.createStringMap(props, "x");
+      expect(stringMap).toEqual({ red: 1, green: 2, blue: 3 });
+    });
+
+    [dataTest, immutableDataTest].forEach(({ createData, testLabel }) => {
+      describe(`returning string maps with ${testLabel}`, () => {
+        const data = createData([
+          { x: "one", y: 1 },
+          { x: "red", y: 2 },
+          { x: "cat", y: 3 }
+        ]);
+
+        it("returns a string map from strings in data", () => {
+          const props = { data };
+          const stringMap = Data.createStringMap(props, "x");
+          expect(stringMap).toEqual({ one: 1, red: 2, cat: 3 });
+        });
+
+        it("a unique set of values is returned from multiple sources", () => {
+          const props = { tickValues, data };
+          const stringMap = Data.createStringMap(props, "x");
+          expect(stringMap).toEqual({
+            one: 1,
+            two: 2,
+            three: 3,
+            red: 4,
+            cat: 5
+          });
+        });
+      });
+    });
+  });
+
+  [dataTest, immutableDataTest].forEach(({ createData, testLabel }) => {
```

Comment:
This is the same pattern I commented on @heythisispaul 's PR - in the future I would like to avoid creating `describe` blocks inside a `forEach`, but for now we'll keep it while we copy over these tests.